mobile/src/main/java/rs/readahead/washington/mobile/views/fragment/forms/DraftFormsListFragment.kt

```diff
@@ -0,0 +1,167 @@
+    private var model: SharedFormsViewModel? = null
+    private var adapter: CollectDraftFormInstanceRecycleViewAdapter? = null
+    private var presenter: CollectFormInstanceListPresenter? = null
+    override fun getFormListType(): FormListInterfce.Type {
+        return FormListInterfce.Type.DRAFT
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        adapter = CollectDraftFormInstanceRecycleViewAdapter(this)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        recyclerView = binding!!.draftFormInstances
+        blankFormsInfo = binding!!.blankDraftFormsInfo
+        val mLayoutManager: RecyclerView.LayoutManager = LinearLayoutManager(activity)
+        recyclerView!!.layoutManager = mLayoutManager
+        recyclerView!!.adapter = adapter
+        model = ViewModelProvider(this).get(
+            SharedFormsViewModel::class.java
+        )
+        createPresenter()
+        initObservers()
+        listDraftForms()
+    }
+
+    private fun initObservers() {
+        model!!.onDraftFormInstanceListSuccess.observe(
+            viewLifecycleOwner
+        ) { instances: List<CollectFormInstance> ->
+            onDraftFormInstanceListSuccess(
+                instances
+            )
+        }
+        model!!.onFormInstanceListError.observe(
+            viewLifecycleOwner
+        ) { error: Throwable? ->
+            onFormInstanceListError(
+                error
+            )
+        }
+        model!!.onFormInstanceDeleteSuccess.observe(
+            viewLifecycleOwner,
+            Observer { success: Boolean? ->
+                onFormInstanceDeleted(
+                    success!!
+                )
+            })
+    }
+
+    private fun onFormInstanceDeleted(success: Boolean) {
+        if (success) {
+            DialogUtils.showBottomMessage(
+                activity,
+                getString(R.string.collect_toast_form_deleted),
+                false
+            )
+            listDraftForms()
+        }
+    }
+
+    override fun onDestroy() {
+        destroyPresenter()
+        super.onDestroy()
+    }
+
+    private fun onDraftFormInstanceListSuccess(instances: List<CollectFormInstance>) {
+        blankFormsInfo!!.visibility = if (instances.isEmpty()) View.VISIBLE else View.GONE
+        adapter!!.setInstances(instances)
+    }
+
+    override fun onFormInstanceListSuccess(instances: List<CollectFormInstance?>) {
+        blankFormsInfo!!.visibility = if (instances.isEmpty()) View.VISIBLE else View.GONE
+        adapter!!.setInstances(instances)
+    }
+
+    override fun onFormInstanceListError(error: Throwable?) {
+        Timber.d(error, javaClass.name)
+    }
+
+    fun listDraftForms() {
+        /*if (model != null) {
+            model.listDraftFormInstances();
+        }*/
+        if (presenter != null) {
```

Comment:
We should remove any instance of `Presenter` as we already have a `ViewModel` that holds listDraftFormInstances